src/main/kotlin/io/github/rybalkinsd/kohttp/dsl/context/BodyContext.kt

```diff
@@ -0,0 +1,40 @@
+package io.github.rybalkinsd.kohttp.dsl.context
+
+import io.github.rybalkinsd.kohttp.util.Form
+import io.github.rybalkinsd.kohttp.util.Json
+import okhttp3.MediaType
+import okhttp3.RequestBody
+import java.io.File
+
+
+/**
+ *
+ * @since 0.1.0
+ * @author sergey, alex
+ */
+@HttpDslMarker
+class BodyContext(type: String?) {
+    private val mediaType = type?.let { MediaType.get(it) }
+
+    fun string(content: String): RequestBody = RequestBody.create(mediaType, content)
+    fun file(content: File): RequestBody = RequestBody.create(mediaType, content)
+    fun bytes(content: ByteArray): RequestBody = RequestBody.create(mediaType, content)
+
+    fun json(content: String): RequestBody = JSON.create { content }
+    fun form(content: String): RequestBody = FORM.create { content }
+
+    fun json(init: Json.() -> Unit): RequestBody = JSON.create { Json().also(init).toString() }
+    fun form(init: Form.() -> Unit): RequestBody = FORM.create { Form().also(init).toString() }
+
+    private fun MediaType.create(contentProducer: () -> Any): RequestBody {
+        return when (val content = contentProducer()) {
+            is String -> RequestBody.create(this, content)
+            is File -> RequestBody.create(this, content)
```

Comment:
@shtykh this is unreachable code. The reason is:

    fun file(content: File): RequestBody = RequestBody.create(mediaType, content)
    fun file(content: File): RequestBody = RequestBody.create(mediaType, content)
    fun bytes(content: ByteArray): RequestBody = RequestBody.create(mediaType, content)

All functions invoke `RequestBody.create` method, but `MediaType.create` is not used.
The only usege is for `json { }` and `form { }`

Reply:
fixed